chain/chain/src/chain.rs

```diff
@@ -2370,6 +2370,30 @@ impl Chain {
     pub fn is_chunk_orphan(&self, hash: &CryptoHash) -> bool {
         self.blocks_with_missing_chunks.contains(hash)
     }
+
+    /// Check if can sync with sync_hash
+    pub fn sync_hash_validity(&mut self, sync_hash: &CryptoHash) -> Result<bool, Error> {
+        let head = self.head()?;
+        match self.get_block_header(sync_hash) {
+            Ok(sync_header) => {
+                let sync_header = sync_header.clone();
+                // The Epoch of sync_hash may be either the current one or the previous one
+                if head.epoch_id == *sync_header.epoch_id()
+                    || head.epoch_id == *sync_header.next_epoch_id()
+                {
+                    if self.runtime_adapter.is_next_block_epoch_start(sync_header.prev_hash())? {
+                        self.get_block(sync_hash)?;
```

Comment:
why?

Reply:
Added a comment.

Reply:
How come we have the header but not the block? This can only happen if we are syncing as well.

Reply:
added `debug_assert!(self.get_block(sync_hash).is_ok())`